proxy/http/HttpSM.cc

```diff
@@ -5996,6 +5996,8 @@ HttpSM::do_setup_post_tunnel(HttpVC_t to_vc_type)
       tunnel.add_consumer(server_entry->vc, ua_entry->vc, &HttpSM::tunnel_handler_post_server, HT_HTTP_SERVER, "http server post");
     }
     server_entry->in_tunnel = true;
+    ink_assert(server_entry->vc != nullptr);
+    server_entry->vc->make_tunnel_endpoint();
```

Comment:
`server_entry->vc` is a `VConnection` pointer, which is why `make_tunnel_endpoint()` has to be virtual member function of `VConnection`.